### PR TITLE
Code cleanups, aligning, also post spec fix

### PR DESCRIPTION
**Pull request by tanraya**
Just did some cleanups for models and also fixed failing post spec.
I'll be happy if you like my changes and merge them with master :)

### CODE REVIEW

File: app/models/forem/topic.rb [hunk 1]
```diff
@@ -101,7 +99,8 @@ def moderate!(option)
 
     # A Topic cannot be replied to if it's locked.
     def can_be_replied_to?
-      !locked?
+      #!locked?
+      not locked?
```
Comment (parndt) [inline app/models/forem/topic.rb]: why?

File: spec/models/post_spec.rb [hunk 2]
```diff
@@ -55,11 +55,11 @@
       topic = FactoryGirl.create(:topic)
       post1 = FactoryGirl.create(:post, :topic => topic)
       topic.reload
-      topic.last_post_at.should == post1.created_at
+      topic.last_post_at.to_s.should == post1.created_at.to_s
 
       post2 = FactoryGirl.create(:post, :topic => topic)
       topic.reload
-      topic.last_post_at.should == post2.created_at
+      topic.last_post_at.to_s.should == post2.created_at.to_s
```
Comment (parndt) [inline spec/models/post_spec.rb:62]: why?

File: app/models/forem/topic.rb (hunk 1 above)
Comment (tanraya) [inline app/models/forem/topic.rb]: This is because this form is closer to natural language

File: spec/models/post_spec.rb (hunk 2 above)
Comment (tanraya) [inline spec/models/post_spec.rb:62]: I had the situation that well described here http://blog.tddium.com/2011/08/07/rails-time-comparisons-devil-details-etc/

If I remove to_s from test then I get:

```
Failures:

  1) Forem::Post after creation sets topics last_post_at value
     Failure/Error: topic.last_post_at.should == post1.created_at
       expected: Fri, 20 Apr 2012 05:16:39 UTC +00:00
            got: Fri, 20 Apr 2012 05:16:39 UTC +00:00 (using ==)
       Diff:
     # ./spec/models/post_spec.rb:58:in `block (3 levels) in <top (required)>'

Finished in 25.52 seconds
168 examples, 1 failure

Failed examples:

rspec ./spec/models/post_spec.rb:54 # Forem::Post after creation sets topics last_post_at value
```

File: app/models/forem/topic.rb (hunk 1 above)
Comment (radar) [inline app/models/forem/topic.rb]: `!locked?` is better imo

Comment (tanraya) [inline app/models/forem/topic.rb]: Should I bring it back?

Comment (radar) [inline app/models/forem/topic.rb]: Yes please.

Comment (tanraya) [inline app/models/forem/topic.rb]: Done

File: spec/models/post_spec.rb (hunk 2 above)
Comment (knewter) [inline spec/models/post_spec.rb:62]: Yeah, you can see this failure happening in forem's travis here: http://travis-ci.org/#!/radar/forem/jobs/1121406

+1 for this change

**Comment by knewter** [comment]
Looks great, all tests passed.  Merged.

**Pull request merged**